src/main/java/com/cloudant/client/internal/views/PageMetadata.java

```diff
@@ -30,23 +43,79 @@
     /**
      * The parameters needed to request the page described by this metadata
      */
+    @Expose
+    @SerializedName("v")
     ViewQueryParameters<K, V> pageRequestParameters;
 
     /**
      * The page number of the page to retrieve
      */
+    @Expose
+    @SerializedName("n")
     long pageNumber;
     /**
      * Indicate the paging direction for this metadata
      */
+    @Expose
+    @SerializedName("d")
     PagingDirection direction;
 
     enum PagingDirection {
+        @SerializedName("F")
         FORWARD,
+        @SerializedName("B")
         BACKWARD
     }
 
     PageMetadata(PagingDirection direction) {
         this.direction = direction;
     }
+
+    static <K, V> PageMetadata<K, V> fromString(final ViewQueryParameters<K, V> initialParameters,
+                                                String base64encoded) {
+
+        String json = new String(Base64.decodeBase64(base64encoded), Charset.forName("UTF-8"));
+
+        //get the type for ViewQueryParameters
+        Type thisType = new TypeToken<ViewQueryParameters<K, V>>() {
+        }.getType();
+
+        // get a suitable Gson, we need an InstanceCreator for ViewQueryParameters, as well as
+        // any adapter registered for the K key type
+        Gson parameterGson = registerKAdapter(initialParameters, new GsonBuilder())
+                .registerTypeAdapter(thisType, new InstanceCreator<ViewQueryParameters<K, V>>() {
+                    @Override
+                    public ViewQueryParameters<K, V> createInstance(Type type) {
+                        return new ViewQueryParameters<K, V>(initialParameters);
+                    }
+                })
+                .create();
+
+        PageMetadata<K, V> pm = parameterGson.fromJson(json, new TypeToken<PageMetadata<K, V>>() {
+        }.getType());
+        //set any parameters that weren't from the encoded string from the initial parameters
+        pm.pageRequestParameters.setNonPagingParametersFrom(initialParameters);
+        return pm;
+    }
+
+    @Override
+    public String toString() {
+        try {
+            Gson g = registerKAdapter(pageRequestParameters, new GsonBuilder())
+                    .excludeFieldsWithoutExposeAnnotation()
+                    .create();
+            return new String(Base64.encodeBase64URLSafe(g.toJson(this).getBytes("UTF-8")),
+                    Charset.forName("UTF-8"));
+        } catch (UnsupportedEncodingException e) {
+            //all JVMs should support UTF-8
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
`toString` should be returning a human-readable string for printing. I'd suggest another name be used.